lib/sender_link.js

```diff
@@ -71,7 +71,15 @@ SenderLink.prototype.send = function(msg, options) {
 
   var self = this;
   return new Promise(function(resolve, reject) {
-    var message = new M.Message(options, self.policy.encoder(msg));
+    var message;
+    if (_.isObject(msg) && msg.hasOwnProperty('body')) {
+      msg.body = self.policy.encoder(msg.body);
```

Comment:
side-effect. maybe instantiate the message, _then_ encode the body in message and not msg?